utils/runtime-fuzzer/src/generator/upload_program.rs

```diff
@@ -100,17 +102,45 @@ fn arbitrary_salt(u: &mut Unstructured) -> Result<Vec<u8>> {
 }
 
 fn config(
+    unstructured: &mut Unstructured,
     programs: Option<&NonEmpty<ProgramId>>,
     codes: Option<&NonEmpty<CodeId>>,
     log_info: Option<String>,
     current_balance: Balance,
-) -> StandardGearWasmConfigsBundle {
+) -> RandomizedGearWasmConfigBundle {
+    // Observation: with balance increased and no control instructions we get *a lot* of programs executed even if we're
+    // running fuzzer for just 5 minutes. Without balance increase disabling control results in gas limit exceeding quite often.
+    let no_control = unstructured.ratio(1, 4).unwrap();
+    // Carefully adjusted to run fine with `account.rs` balance calculation.
+    // 1) When max_balance or max_balance/4 is used we're almost always guaranteed to finish execution of a program so
+    // do not try to get too much instructions in case of control insns enabled to not get infinite loops or recursion.
+    // 2) Otherwise we can run quite a lot of insns
+    // with no control as we're guaranteed to terminate execution.
+    let max_instructions = if no_control {
+        // when no control insns are enabled we generate a small amount of instructions
+        // as it should be more than enough to test the program and exhaust the gas
+        unstructured.int_in_range(80..=400).unwrap()
+    } else {
+        unstructured.int_in_range(500..=1200).unwrap()
+    };
+
+    let (min_funcs, max_funcs) = no_control.then(|| (1, 1)).unwrap_or((3, 4));
+
     let initial_pages = 2;
-    let mut injection_types = SyscallsInjectionTypes::all_with_range(1..=3);
+    // pump up injection rates of syscalls when there's control instructions and lower it when there's no control
```

Comment:
I think of setting "wait*" injection range to `0..=0` in case of `no controls`. Almost all executions of `no-control` then must go to wait list, because you have only 1 functions with < 400 instructions, i.e. chance of being in a waitlist if `gr_wait*` is inserted is ~100%